core/src/main/java/com/yahoo/oak/EntryHashSet.java

```diff
@@ -0,0 +1,561 @@
+ *   |                        | access the key off-heap                         | entry with
+ * -----------------------------------------------------------------------------| entry hash index
+ * 1 | Value Reference        | Reference encoding all info needed to           | 0
+ *   |                        | access the key off-heap.                        |
+ *   |                        | The encoding of key and value can be different  |
+ * -----------------------------------------------------------------------------|
+ * 2 | Hash  - FULL entry hash index, which might be different (bigger) than    |
+ *   |         entry hash index                                                 |
+ * ---------------------------------------------------------------------------------------
+ * 3 | Key Reference          | Deleted reference (still) encoding all info     |
+ *   |                        | needed to access the key off-heap               |
+ * -----------------------------------------------------------------------------| Deleted entry with
+ * 4 | Value Reference        | Deleted reference(still)  encoding all info     | entry hash index
+ *   |                        | needed to access the key off-heap.              | 1
+ * -----------------------------------------------------------------------------|
+ * 5 | Hash  - FULL entry hash index, which might be different (bigger) than    |
+ *   |         entry hash index                                                 |
+ * ---------------------------------------------------------------------------------------
+ * 6 | Key Reference          | Invalid key reference                           | empty entry
+ * -----------------------------------------------------------------------------|
+ * 7 | Value Reference        | Invalid value reference                         |
+ * -----------------------------------------------------------------------------|
+ * 8 | 000...00                                                                 |
+ * ---------------------------------------------------------------------------------------
+ * ...
+ *
+ *
+ * Internal class, package visibility
+ */
+class EntryHashSet<K, V> extends EntryArray<K, V> {
+
+    /*-------------- Constants --------------*/
+    static final int INVALID_FULL_HASH = 0; // because memory is initially zeroed
+    static final int DEFAULT_COLLISION_CHAIN_LENGTH = 3;
+    // HASH - the full hash index of this entry (one integer, all bits).
+    private static final int HASH_FIELD_OFFSET = 2;
+    private static final int FULL_HASH_BITS = 32; // Hash needs to be an integer
+    // One additional to the key and value references field, which keeps the full hash index of the entry
+    private static final int ADDITIONAL_FIELDS = 1;  // # of primitive fields in each item of entries array
+
+    // number of entries candidates to try in case of collision
+    private AtomicInteger collisionChainLength = new AtomicInteger(DEFAULT_COLLISION_CHAIN_LENGTH);
```

Comment:
Intellij: Field 'collisionChainLength' may be 'final'

Reply:
I do not get this message. How it can be final if it is updated with `collisionChainLength.incrementAndGet();` ? Not fixed